iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1478,6 +1465,8 @@ pub struct DirectAddrInfo {
     pub last_control: Option<(Duration, ControlMsg)>,
     /// How long ago was the last payload message for this node.
     pub last_payload: Option<Duration>,
+    /// When was this connection last alive, if ever.
+    pub last_alive: Option<Duration>,
```

Comment:
This might be redundant, tldr this is the max between the two above. But it's harmless anyway